File: lang/c++/test/vac.fun/data_structures/LinkedStackTest.cpp
#include <vac.fun/data_structures/LinkedStack.hpp>
#include <vac.fun/util/TestRunner.hpp>

using namespace std;

bool testLinkedStack() {
    size_t size = 8192;

    auto stack = new LinkedStack::LinkedStack<int>();

    for (size_t i = 1; i <= size; i++) {
        stack->push(static_cast<int>(i));
    }

    if (stack->size() != size) {
        return false;
    }

    for (size_t i = size; i > 0; i--) {
        if (static_cast<size_t>(stack->peek()) != i) {
            return false;
        }
        stack->pop();
    }

    return stack->isEmpty();
}

int main() {
    TestRunner::parseTest(testLinkedStack());
}

File: lang/php/test/vac.fun/data_structures/LinkedStackTest.php
<?php
    require_once "vac.fun/data_structures/LinkedStack.php";
    require_once "vac.fun/util/TestRunner.php";

    use vac\fun\data_structures\LinkedStack\LinkedStack;
    use vac\fun\util\TestRunner;

    function testLinkedStack(): bool {
        $size = 8192;

        $stack = new LinkedStack();

        for ($i = 1; $i <= $size; $i++) {
            $stack->push($i);
        }

        if ($stack->size() != $size) {
            return false;
        }

        for ($i = $size; $i > 0; $i--) {
            if ($stack->peek() != $i) {
                return false;
            }
            $stack->pop();
        }

        return $stack->isEmpty();
    }

    if (count(debug_backtrace()) == 0) {
        TestRunner\parseTest(testLinkedStack());
    }
?>
